src/main/java/ru/evotor/framework/receipt/formation/event/ReturnPositionsForBarcodeRequestedEvent.kt

```diff
@@ -12,29 +13,45 @@ import ru.evotor.framework.receipt.Position
  * Обрабатывая данные, содержащиеся в событии, приложения могут добавлять позиции в чек и / или создавать новые товары.
  *
  * @property barcode строка данных, полученных от сканера штрихкодов.
+ * @property extractedData агрегированные данные, которые удалось получить из отсканированного штрихкода
  * @property creatingNewProduct указывает на необходимость создать новый товар. Сразу после сканирования штрихкода всегда содержит false.
  * @see <a href="https://developer.evotor.ru/docs/doc_java_return_positions_for_barcode_requested.html">Обработка события сканирования штрихкода</a>
  */
 data class ReturnPositionsForBarcodeRequestedEvent(
         val barcode: String,
+        val extractedData: DataExtracted?,
         val creatingNewProduct: Boolean
 ) : IntegrationEvent() {
 
     override fun toBundle() = Bundle().apply {
         putString(KEY_BARCODE_EXTRA, barcode)
+        putBundle(KEY_EXTRACTED_DATA_EXTRA, extractedData?.toBundle())
         putBoolean(KEY_CREATE_PRODUCT_EXTRA, creatingNewProduct)
     }
 
     companion object {
         const val KEY_BARCODE_EXTRA = "key_barcode_extra"
+        const val KEY_EXTRACTED_DATA_EXTRA = "key_extracted_data_extra"
         const val KEY_CREATE_PRODUCT_EXTRA = "key_create_product_extra"
 
         @JvmStatic
         fun from(bundle: Bundle?) = bundle?.let {
+            if (!it.containsKey(KEY_EXTRACTED_DATA_EXTRA)) {
+                ReturnPositionsForBarcodeRequestedEvent(
+                        barcode = it.getString(KEY_BARCODE_EXTRA)
+                                ?: return null,
+                        extractedData = null,
+                        creatingNewProduct = it.getBoolean(KEY_CREATE_PRODUCT_EXTRA)
+                )
```

Comment:
ReturnPositionsForBarcodeRequestedEvent(
                        barcode = it.getString(KEY_BARCODE_EXTRA)
                                ?: return null,
                        extractedData = if (!it.containsKey(KEY_EXTRACTED_DATA_EXTRA)){null} else{DataExtracted.from(
                            it.getBundle(KEY_EXTRACTED_DATA_EXTRA)
                    )},
                        creatingNewProduct = it.getBoolean(KEY_CREATE_PRODUCT_EXTRA)
                )

Reply:
done